### DISCUSSION TITLE
Stream raw image matrix

### DISCUSSION THREAD

**Discussion by kipcole9** [akash-akya/vix#148] [Q&A]
@akash-akya, I’m answering [this question on Elixir forum](https://elixirforum.com/t/how-to-list-all-i-j-coordinates-for-cells-equal-to-1/62924) using Vix/Image because its a fun problem which is trivial with `Operation.xyz/3` - except I don’t think we can currently stream raw image data, only formatted images, with `Image.write_to_stream/2`.

Do you see any possibility of streaming the raw pixel data, rather than the formatting image?

PS: I know `Vix` is not the optimal solution to the problem as stated, but since there are no answers after a few days I think its cool to show that Vix/Image can do this - albeit the whole image needs to be rendered in memory, not streamed unless you know a way.

**Comment by akash-akya** [comment]
Hi @kipcole9, I had thought of this but I was not sure of concrete use case then. One concern I had was that caller has to be aware of how to interpret the stream of binary blob (shape, bands, format etc). Anyway I not sure if that is a big concern given that we already have `write_to_binary` anyway. I'll look into this.
